addon/components/x-xml-http-request.js

```diff
@@ -6,34 +6,67 @@ import XRequest from 'x-request/x-request';
 export default Ember.Component.extend({
   layout: layout,
 
-  'data': '',
   'method': 'POST',
   'response-type': 'json',
   'with-credentials': false,
   'timeout': 0,
   'url': '',
   'headers': Ember.computed(function() { return {}; }),
 
-  generateRequest: Ember.observer('method', 'response-type', 'with-credentials', 'url', 'data', 'headers', function() {
-    this.request = new XRequest({
+  'request-constructor': XRequest,
+
+  'resetCount': 0,
+
+  request: Ember.computed('method', 'response-type', 'with-credentials', 'url', 'headers', 'timeout', 'resetCount', function() {
+    let Request = this.get('request-constructor');
+
+    let request = new Request({
       freeze: false,
       responseType: this.get('response-type'),
       withCredentials: this.get('with-credentials'),
       timeout: this.get('timeout'),
       observe: Ember.run.bind(this, function(state) {
-        this.set('model', state);
+        this.set('currentRequestState', state);
       })
     });
-    this.request.open(this.get('method'), this.get('url'));
     let headers = this.get('headers') || {};
     Object.keys(headers).forEach((key)=> {
-      this.request.setRequestHeader(key, headers[key]);
+      request.setRequestHeader(key, headers[key]);
+    });
+    return request;
+  }),
+
+  setInitialState: Ember.observer('request', function() {
+    this.set('currentRequestState', this.get('request.state'));
+  }),
+
+  model: Ember.computed('currentRequestState', function() {
+    let request = this.get('request');
+    let config = this.getProperties('method', 'url');
+    let state = this.get('currentRequestState') || this.get('request.state');
+    let component = this;
+    return Object.create(state, {
+      send: {
+        value: function(data) {
+          request.open(config.method, config.url);
+          request.send(data);
+        }
+      },
+      abort: {
+        value:  function() {
+          request.abort();
+        }
+      },
+      reset: {
+        value: function() {
+          component.incrementProperty('resetCount');
+        }
+      }
```

Comment:
Meta point: Do you think this will be a tripping point for ember newbies looking for the actions hash?